app/src/main/java/com/example/polina/socialnetwork/FormActivity.java

```diff
@@ -1,39 +1,252 @@
+
+        if (requestCode == RESULT_LOAD_IMAGE && resultCode == RESULT_OK && null != data) {
+            selectedImage = data.getData();
+            twimage.setImageURI(selectedImage);
+        }
+    }
 
-        //noinspection SimplifiableIfStatement
-        if (id == R.id.action_settings) {
-            return true;
+    public String getRealPathFromURI(Context context, Uri contentUri) {
+        Cursor cursor = null;
+        try {
+            String[] proj = {MediaStore.Images.Media.DATA};
+            cursor = context.getContentResolver().query(contentUri, proj, null, null, null);
+            int column_index = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
+            cursor.moveToFirst();
+            return cursor.getString(column_index);
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
         }
+    }
+
+    public void submit(View v) {
+        name = twname.getText().toString();
+        birthday = twbithday.getText().toString();
+        sex = twsex.getText().toString();
 
-        return super.onOptionsItemSelected(item);
+        new AsyncTask<Void, Void, List<String>>() {
+
+            @Override
+            protected List<String> doInBackground(Void... voids) {
+                String path = getRealPathFromURI(FormActivity.this, selectedImage);
+                InputStream inputStream = getThumbnailImage(selectedImage);
+                String url = ConnectionTo3S.uploadImage(path);
+                String urlSmall = ConnectionTo3S.uploadImage(inputStream);
+                List<String> listOfUrls = new ArrayList<>();
+                listOfUrls.add(url);
+                listOfUrls.add(urlSmall);
+                return listOfUrls;
+            }
+
+            @Override
+            protected void onPostExecute(List<String> urls) {
+                if (urls != null) {
+                    image = urls.get(0);
+                    imagemini = urls.get(1);
+                }
+            }
+        }.execute();
+
+        new AsyncTask<Void, Void, Void>() {
+
+            @Override
+            protected Void doInBackground(Void... voids) {
+                try {
+                    JSONObject o = ((SNApp) getApplication()).api.userInfo(name, birthday, sex, image, imagemini, FormActivity.this);
+                    if(o!= null){
+                        intent = new Intent(FormActivity.this, MainActivity.class);
+                        startActivity(intent);
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return null;
+            }
+
+        }.execute();
+    }
+
+
+    public InputStream getThumbnailImage(Uri uri) {
+        try {
+            Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), uri);
+            System.out.println(bitmap.toString());
+            Bitmap resizedBitmap = getResizedBitmap(bitmap, 100, 100);
```

Comment:
100 - should be constant http://en.wikipedia.org/wiki/Magic_number_%28programming%29